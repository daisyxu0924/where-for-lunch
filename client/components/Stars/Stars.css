@import 'styles/variables.css';

.root {
  display: inline-block;
}

.stars {
  display: inline;
}
.stars {
  position: relative;
  vertical-align: middle;
  display: inline-block;
  color: $starGrey;
  overflow: hidden;
  margin-left: calc($base-spacing / 2);
}
.fullStars {
  position: absolute;
  left: 0;
  top: 0;
  white-space: nowrap;
  overflow: hidden;
  color: $starYellow;
}
.emptyStars:before, .fullStars:before {
  content:'\2605\2605\2605\2605\2605';
  font-size: 1rem;
}