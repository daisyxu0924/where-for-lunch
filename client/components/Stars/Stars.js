import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Stars.css';

class Stars extends PureComponent {
  static propTypes = {
    rating: PropTypes.number,
  };

  render() {
    const { rating } = this.props;
    return (
      <span className={styles.root}>
        {
          rating &&
          <div className={styles.rating}>
            <div className={styles.stars}>
              <div className={styles.emptyStars}></div>
              <div className={styles.fullStars} style={{ width: `${rating / 5 * 100}%` }}></div>
            </div>
          </div>
        }
      </span>
    );
  }
}

export default Stars;
