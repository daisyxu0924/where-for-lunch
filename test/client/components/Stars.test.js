import React from 'react';
import Stars from 'client/components/Stars/Stars';
import renderer from 'react-test-renderer';

describe('Stars Component', () => {
  test('renders properly', () => {
    const placeComponent = renderer
      .create(<Stars rating={4} />)
      .toJSON();
    expect(placeComponent).toMatchSnapshot();
  });
});
