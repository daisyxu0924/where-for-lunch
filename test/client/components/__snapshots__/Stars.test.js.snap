// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars Component renders properly 1`] = `
<span
  className="root"
>
  <div
    className="rating"
  >
    <div
      className="stars"
    >
      <div
        className="emptyStars"
      />
      <div
        className="fullStars"
        style={
          Object {
            "width": "80%",
          }
        }
      />
    </div>
  </div>
</span>
`;
